From 981bfca72414bfa7a97b6526e4736ea4f6834620 Mon Sep 17 00:00:00 2001
From: Rajasekar Raja <rajasekarr@nvidia.com>
Date: Mon, 22 Apr 2024 13:50:47 -0400
Subject: [PATCH] From a88498262d8d88fb3846d1a5c1a373751fe6f381 Mon Sep 17
 00:00:00 2001 Subject: [PATCH 11/11] zebra: Actually display I/O buffer sizes
 (part-2)

An extension of commit-8d8f12ba8e5cd11c189b8475b05539fa8415ccb9

Removing ifdef DEV_BUILD in stream_fifo_push as well to make the 'sh
zebra client' display the current I/O fifo along with max fifo items.

TICKET :#3390099

Signed-off-by: Rajasekar Raja <rajasekarr@nvidia.com>

diff --git a/lib/stream.c b/lib/stream.c
index 2de3abdf45..3aef70794e 100644
--- a/lib/stream.c
+++ b/lib/stream.c
@@ -1256,9 +1256,7 @@ void stream_fifo_init(struct stream_fifo *fifo)
 /* Add new stream to fifo. */
 void stream_fifo_push(struct stream_fifo *fifo, struct stream *s)
 {
-#if defined DEV_BUILD
 	size_t max, curmax;
-#endif
 
 	if (fifo->tail)
 		fifo->tail->next = s;
@@ -1267,15 +1265,11 @@ void stream_fifo_push(struct stream_fifo *fifo, struct stream *s)
 
 	fifo->tail = s;
 	fifo->tail->next = NULL;
-#if !defined DEV_BUILD
-	atomic_fetch_add_explicit(&fifo->count, 1, memory_order_release);
-#else
 	max = atomic_fetch_add_explicit(&fifo->count, 1, memory_order_release);
 	curmax = atomic_load_explicit(&fifo->max_count, memory_order_relaxed);
 	if (max > curmax)
 		atomic_store_explicit(&fifo->max_count, max,
 				      memory_order_relaxed);
-#endif
 }
 
 void stream_fifo_push_safe(struct stream_fifo *fifo, struct stream *s)
-- 
2.43.2

